lib/src/Pos/DB/Block.hs

```diff
@@ -289,192 +155,103 @@ decodeOrFailPureDB
     -> Either Text (Block, Undo)
 decodeOrFailPureDB = decodeFull
 
-dbGetBlundPure ::
+dbGetBlundPureDefault ::
        (HasConfiguration, MonadPureDB ctx m)
     => HeaderHash
     -> m (Maybe (Block, Undo))
-dbGetBlundPure h = do
+dbGetBlundPureDefault h = do
     (blund :: Maybe ByteString) <-
         view (pureBlocksStorage . at h) <$> (view (lensOf @DBPureVar) >>= readIORef)
     case decodeOrFailPureDB <$> blund of
         Nothing        -> pure Nothing
         Just (Left e)  -> throwM (DBMalformed e)
         Just (Right v) -> pure (Just v)
 
-dbGetBlockPureDefault ::
-       (HasConfiguration, MonadPureDB ctx m)
-    => HeaderHash
-    -> m (Maybe Block)
-dbGetBlockPureDefault h = fmap fst <$> dbGetBlundPure h
-
-dbGetUndoPureDefault ::
-       forall ctx m. (HasConfiguration, MonadPureDB ctx m)
+dbGetRawBlockPureDefault
+    :: (HasConfiguration, MonadPureDB ctx m)
     => HeaderHash
-    -> m (Maybe Undo)
-dbGetUndoPureDefault h = fmap snd <$> dbGetBlundPure @ctx @m h
+    -> m (Maybe RawBlock)
+dbGetRawBlockPureDefault h = fmap (RawBlock . serialize' . fst) <$> dbGetBlundPureDefault h
 
-dbGetHeaderPureDefault ::
-       (HasConfiguration, MonadDBRead m)
+dbGetRawUndoPureDefault
+    :: forall ctx m. (HasConfiguration, MonadPureDB ctx m)
     => HeaderHash
-    -> m (Maybe BlockHeader)
-dbGetHeaderPureDefault = blkGetHeader
+    -> m (Maybe RawUndo)
+dbGetRawUndoPureDefault h = fmap (RawUndo . serialize' . snd) <$> dbGetBlundPureDefault @ctx @m h
 
-dbPutBlundPureDefault ::
+dbPutRawBlundPureDefault ::
        forall ctx m. (HasConfiguration, MonadPureDB ctx m)
-    => Blund
+    => RawBlund
     -> m ()
-dbPutBlundPureDefault (blk,undo) = do
+dbPutRawBlundPureDefault (blk, rawUndo) = do
+    let blund :: Blund
+        blund =
+            either (error "Couldn't deserialize undo")
+                   (blk,)
+                   (decodeFull rawUndo)
```

Comment:
It's absolutely weird to decode and then immediately encode. I suggest that we store blocks and undos separately in pure db. Can you create an issue at least?

Reply:
>I suggest that we store blocks and undos separately in pure db.

we don't, why do you pay attention to it at all? :thinking: 
I don't think it does matter how we store blund in pure db

Reply:
> we don't

I know that we don't, but I suggest that we do.

> why do you pay attention to it at all?

Because doing unnecessary roundtrip of encoding and decoding is inefficient and it will make tests slower. So I would like to have an issue about it.